src/adapter/oauth/src/oauth_github.rs

```diff
@@ -19,12 +22,19 @@ const LOGIN_METHOD_GITHUB: &str = "oauth_github";
 
 ///////////////////////////////////////////////////////////////////////////////
 
-pub struct OAuthGithub;
+pub struct OAuthGithub {
+    cached_accounts_by_name: Mutex<HashMap<String, GithubAccountInfo>>,
```

Comment:
I'd recommend to create a private struct called `State` with two hash maps, and then wrap it into a single mutex.

This way we can only lock once per update and be sure that we don't deadlock.

Reply:
Addressed